server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -210,39 +292,56 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        final String level = params.param("level", "node");
-        final boolean isLevelValid = "indices".equalsIgnoreCase(level)
-            || "node".equalsIgnoreCase(level)
-            || "shards".equalsIgnoreCase(level);
+        final String level = params.param("level", Fields.NODE);
+        final boolean isLevelValid = Fields.NODE.equalsIgnoreCase(level)
+            || Fields.INDICES.equalsIgnoreCase(level)
+            || Fields.SHARDS.equalsIgnoreCase(level);
         if (!isLevelValid) {
-            throw new IllegalArgumentException("level parameter must be one of [indices] or [node] or [shards] but was [" + level + "]");
+            throw new IllegalArgumentException(
+                "level parameter must be one of ["
+                    + Fields.INDICES
+                    + "] or "
+                    + "["
+                    + Fields.NODE
+                    + "] or ["
+                    + Fields.SHARDS
+                    + "] but was ["
+                    + level
+                    + "]"
+            );
         }
 
         // "node" level
         builder.startObject(Fields.INDICES);
         stats.toXContent(builder, params);
 
-        if ("indices".equals(level)) {
-            Map<Index, CommonStats> indexStats = createStatsByIndex();
+        if (Fields.INDICES.equals(level)) {
+            if (statsByIndex == null && statsByShard != null) {
+                statsByIndex = createStatsByIndex(statsByShard);
```

Comment:
Added this because since we anyways have `statsByShards`, we can still generate the `statsByIndices`. This is ensure that the behaviour does not break if the response is coming from an older version node which does not support sending `statsByIndex`. In such cases, we just go back to the original behaviour and generate `statsByIndex` right before the building the `ToXContent`.